samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ConfirmPublishAssessmentListener.java

```diff
@@ -444,6 +449,31 @@ public void processAction(ActionEvent ae) throws AbortProcessingException {
         }
     }
 
+    String secureDeliveryModuleId = assessmentSettings.getSecureDeliveryModule();
+    if (StringUtils.equals(secureDeliveryModuleId, SecureDeliverySeb.MODULE_NAME)) {
+        boolean sebFileUploadError = false;
+        Iterator<SectionFacade> sectionFacadeIterator = assessment.getSectionSet().iterator();
+        while (sectionFacadeIterator.hasNext()){
+            SectionFacade sectionFacade = sectionFacadeIterator.next();
+            Iterator<ItemFacade> itemFacadeIterator = sectionFacade.getItemFacadeSet().iterator();
+            while (itemFacadeIterator.hasNext()){
+                ItemFacade itemFacade = itemFacadeIterator.next();
+                if (TypeIfc.FILE_UPLOAD.equals(itemFacade.getType().getTypeId())) {
+                    String sebUpload_err= ContextUtil.getLocalizedString("org.sakaiproject.tool.assessment.bundle.AssessmentSettingsMessages" , "seb_file_upload_error");
+                    context.addMessage(null,new FacesMessage(sebUpload_err));
+                    sebFileUploadError=true;
+                    break;
+                }
+            }
+            if (sebFileUploadError) {
+                break;
+            }
+        }
+        if (sebFileUploadError) {
+            error = true;
+        }
+    }
```

Comment:
Here is an improved block generated by AI
```suggestion
if (StringUtils.equals(assessmentSettings.getSecureDeliveryModule(), SecureDeliverySeb.MODULE_NAME)) {
    boolean sebFileUploadError = false;
    
    for (SectionFacade sectionFacade : assessment.getSectionSet()) {
        for (ItemFacade itemFacade : sectionFacade.getItemFacadeSet()) {
            if (TypeIfc.FILE_UPLOAD.equals(itemFacade.getType().getTypeId())) {
                String sebUploadError = ContextUtil.getLocalizedString(
                    "org.sakaiproject.tool.assessment.bundle.AssessmentSettingsMessages", 
                    "seb_file_upload_error"
                );
                context.addMessage(null, new FacesMessage(sebUploadError));
                sebFileUploadError = true;
                break;
            }
        }
        if (sebFileUploadError) {
            break;
        }
    }
    if (sebFileUploadError) {
        error = true;
    }
}
```